src/bashi/results.py

```diff
@@ -641,3 +641,23 @@ def filter_func(param_val_pair: ParameterValuePair) -> bool:
         return True
 
     parameter_value_pairs[:] = filter(filter_func, parameter_value_pairs)
+
+
+def _remove_unsupported_gcc_versions_for_ubuntu2004(
+    parameter_value_pairs: List[ParameterValuePair],
+):
+    """Remove pairs where GCC version 6 and older is used with Ubuntu 20.04.
+
+    Args:
+        parameter_value_pairs (List[ParameterValuePair]): List of parameter-value pairs.
+    """
+    for compiler_type in (HOST_COMPILER, DEVICE_COMPILER):
+        remove_parameter_value_pairs(
+            parameter_value_pairs,
+            parameter1=compiler_type,
+            value_name1=GCC,
+            value_version1=">6",
+            parameter2=UBUNTU,
+            value_name2=UBUNTU,
+            value_version2="==20.04",
+        )
```

Comment:
```suggestion
for compiler_type in (HOST_COMPILER, DEVICE_COMPILER):
        for gcc_version in range(1, 7):
            remove_parameter_value_pairs(
                parameter_value_pairs,
                removed_parameter_value_pairs,
                parameter1=compiler_type,
                value_name1=GCC,
                value_version1=gcc_version,
                parameter2=UBUNTU,
                value_name2=UBUNTU,
                value_version2="<20.04",
            )
```

Is not doable with two version ranges. Fortunately, released GCC versions are countable. Therefore we can remove each of the version by hand.